### PR TITLE
Smartswitch support for container checker and frr

### PR DESCRIPTION
**Pull request by rameshraghupathy**
#### Why I did it
Some sonic-mgmt container-checker tests fail on the smartswitch platform as the containers running on smartswitch are different than the fixed chassis. Similarly some frr tests fail on smartswitch platform as we have chassisdb in smartswitch, the frr code takes the wrong code path related to modular chassis.

Note: #18676 was created first and however that was created using 202311 branch and this logic is not needed there.

#### How I did it
Created a function is_smartswitch() similart to is_supervisor() using the presence of "DPUS" in platform.json file. Used this function to differentiate smatswitch chassis from modular chassis.  Similarly in frr used the same logic to create an env variable to differentiate smatswitch chassis from modular chassis.

#### How to verify it
Run the sonic-mgmt frr tests and any test. They all pass with this fix.

#### Which release branch to backport (provide reason below if selected)
None

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/sonic-py-common/sonic_py_common/device_info.py [hunk 1]
```diff
@@ -576,6 +576,25 @@ def is_chassis():
...
+    platform = get_platform()
+    if not platform:
+        return False
+
+    # get platform.json file path
+    platform_json = os.path.join(HOST_DEVICE_PATH, platform, "platform.json")
+    try:
+        with open(platform_json, 'r') as f:
+            platform_cfg = json.loads(f.read())
+            if "DPUS" in platform_cfg:
```
Comment (oleksandrivantsiv) [inline src/sonic-py-common/sonic_py_common/device_info.py:590]: return "DPUS" in platform_cfg

**Comment by oleksandrivantsiv** [comment]
LGTM

**Review by oleksandrivantsiv — APPROVED** [review]

**Comment by prgeor** [comment]
@rameshraghupathy where is `is_smartswitch()` function as mentioned in PR description?

File: dockers/docker-fpm-frr/base_image_files/TSB [hunk 2]
```diff
@@ -6,7 +6,18 @@ if [ "$EUID" -ne 0 ] ; then
...
+# Parse the platform.json file to get the platform specific information
+PLATFORM_JSON=/usr/share/sonic/device/$PLATFORM/platform.json
+if [ -f "$PLATFORM_JSON" ]; then
+    NUM_DPU=$(jq -r '.DPUS | length' $PLATFORM_JSON 2>/dev/null)
+    if [[ $NUM_DPU -gt 0 ]]; then
+        SMARTSWITCH=true
+    fi
+fi
+
+if [ -f /etc/sonic/chassisdb.conf ] && [ "$SMARTSWITCH" = false ] ; then
```
Comment (prgeor) [inline dockers/docker-fpm-frr/base_image_files/TSB:20]: @rameshraghupathy Why would `/etc/sonic/chassisdb.conf` be present for smartswich platform?

**Review by prgeor — DISMISSED** [review]

File: dockers/docker-platform-monitor/docker-pmon.supervisord.conf.j2 [hunk 3]
```diff
@@ -28,7 +28,7 @@ stdout_logfile=syslog
 stderr_logfile=syslog
 dependent_startup=true
 
-{% if not skip_chassisd and IS_MODULAR_CHASSIS == 1 %}
+{% if include_chassisd or (not skip_chassisd and IS_MODULAR_CHASSIS == 1) %}
```
Comment (prgeor) [inline dockers/docker-platform-monitor/docker-pmon.supervisord.conf.j2:31]: @rameshraghupathy should this be like to make it clear we are making this change fro smart switch?
if not `skip_chassisd` and (IS_MODULAR_CHASSIS == 1 or IS_SMART_SWITCH)`

Comment (rameshraghupathy) [reply dockers/docker-platform-monitor/docker-pmon.supervisord.conf.j2:31]: We can make it  "if IS_SMART_SWITCH or (IS_MODULAR_CHASSIS == 1 and not skip_chassisd)"  We want to have it all the time when it is smartswitch

Comment (rameshraghupathy) [reply dockers/docker-platform-monitor/docker-pmon.supervisord.conf.j2:31]: @prgeor

**Review by prgeor — APPROVED** [review]

**Comment by prsunny** [comment]
@kperumalbfn , can you please merge this?

File: rules/config [hunk 4]
```diff
@@ -150,7 +150,7 @@ INCLUDE_NAT = y
 INCLUDE_DHCP_RELAY = y
 
 # INCLUDE_DHCP_SERVER - build and install dhcp-server package
-INCLUDE_DHCP_SERVER ?= n
+INCLUDE_DHCP_SERVER = y
```
Comment (kperumalbfn) [inline rules/config:153]: Why DHCP server package change? Any additional checks?

File: dockers/docker-fpm-frr/base_image_files/TSC [hunk 5]
```diff
@@ -6,7 +6,18 @@ if [ "$EUID" -ne 0 ] ; then
...
+PLATFORM="$(sonic-cfggen -d -v DEVICE_METADATA.localhost.platform)"
+SMARTSWITCH=false
+# Parse the platform.json file to get the platform specific information
+PLATFORM_JSON=/usr/share/sonic/device/$PLATFORM/platform.json
+if [ -f "$PLATFORM_JSON" ]; then
+    NUM_DPU=$(jq -r '.DPUS | length' $PLATFORM_JSON 2>/dev/null)
+    if [[ $NUM_DPU -gt 0 ]]; then
+        SMARTSWITCH=true
+    fi
+fi
```
Comment (kperumalbfn) [inline dockers/docker-fpm-frr/base_image_files/TSC:18]: Can you move this common code to TS as this is used in TSA/TSB/TSC sripts?

File: rules/config (hunk 4 above)
Comment (rameshraghupathy) [reply rules/config:153]: DHCP server is needed for DPU ip address assignment

Comment (kperumalbfn) [reply rules/config:153]: Yeah, but this change will enable DHCP server for all the platforms. Could you add Smartswitch checks and enable DHCP server?

Comment (rameshraghupathy) [reply rules/config]: Done

Comment (rameshraghupathy) [reply rules/config]: @kperumalbfn

File: dockers/docker-fpm-frr/base_image_files/TSC (hunk 5 above)
Comment (rameshraghupathy) [reply dockers/docker-fpm-frr/base_image_files/TSC]: Done

Comment (rameshraghupathy) [reply dockers/docker-fpm-frr/base_image_files/TSC]: @kperumalbfn

**Review by kperumalbfn — APPROVED** [review]
LGTM

**Pull request merged**